src/openeo_gfmap/fetching/commons.py

```diff
@@ -100,11 +137,11 @@ def _load_collection(
             spatial_extent, BoundingBoxExtent
         ), "Please provide only a bounding box for tile based fetching."
         spatial_extent = dict(spatial_extent)
-        cube = connection.load_collection(
-            collection_id=collection_name,
+        cube = load_collection_method(
+            connection=connection,
+            bands=bands,
             spatial_extent=spatial_extent,
             temporal_extent=temporal_extent,
-            bands=bands,
             properties=load_collection_parameters,
         )
     elif fetch_type == FetchType.POINT:
```

Comment:
We might consider using match cases instead of if/elif for readability: https://docs.python.org/3/tutorial/controlflow.html#match-statements
(might again be out of scope for this PR and should be put in an issue